addons/point_of_sale/models/pos_session.py

```diff
@@ -1875,18 +1875,18 @@ def find_product_by_barcode(self, barcode, config_id):
         packaging_params = {
             'search_params': {
                 'domain': domain,
-                'fields': ['name', 'barcode', 'product_id', 'qty'],
+                'fields': ['name', 'product_id', 'uom_id'],
             },
         }
-        packaging_params['search_params']['domain'] = [['barcode', '=', barcode]]
-        packaging = self.env['product.packaging'].search(packaging_params['search_params']['domain'])
+        packaging_params['search_params']['domain'] = [['name', '=', barcode]]
+        packaging = self.env['product.uom'].search(packaging_params['search_params']['domain'])
         product = packaging.product_id.with_context(product_context)
         condition = packaging and packaging.product_id
         return {
             'product.product': product.read(product_fields, load=False) if condition else [],
             'product.template.attribute.value': product.product_template_attribute_value_ids.read(product_tmpl_attr_value_fields, load=False) if condition else [],
             'product.template': product.product_tmpl_id.read(product_template_fields, load=False) if condition else [],
-            'product.packaging': packaging.read(product_packaging_fields, load=False) if condition else [],
+            'product.uom': packaging.read(packaging_fields, load=False) if condition else [],
```

Comment:
As it's no longer returning the same model (with a different field structure), really need to make sure the JS-side still works properly with different data